### PR TITLE
[Threading] Fix threading package defaults.

### LINKED ISSUE #59568: The stdlib build currently only allows setting a single threading library type, even for multiple stdlibs
**Describe the bug**
Since #59287, it is possible to set a threading library for the Swift stdlib with `SWIFT_THREADING_PACKAGE`, which is [set to the host platform](https://github.com/apple/swift/pull/59287/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR197) by [default](https://github.com/apple/swift/pull/59287/files#diff-5c1eb2d6768a23753402274997290de28b4c44906856c32d858730083df6857eR196). However, [the current config builds all stdlibs with a single type of threading library](https://github.com/apple/swift/blob/f5bdb858e0e5a6476a00ff20db05cf74a0ded49c/cmake/modules/Threading.cmake#L23), which can be overridden with the flag `--swift-threading-package` but still applies to every stdlib. Since it's possible to cross-compile multiple non-host stdlibs, this won't work if any of them need a different threading library, though it's fine if they're all Darwin platforms with the same type of threading library, say macOS and iOS.

**To Reproduce**
1. Build multiple stdlibs that require different threading libraries, and watch it usually fail when all are built against the host threading library type.

**Expected behavior**
You should be able to build multiple stdlibs against different types of threading libraries.

**Environment:**
 - OS: Ubuntu 20.04 x86_64

**Additional context**
I just ran into this because the Android armv7 stdlib built fine on the community Android armv7 CI, but failed on my Android CI. I dug into it and it's because the host SDK on the community CI is `LINUX` so [it applies that threading library to Android armv7 too](https://ci-external.swift.org/job/oss-swift-RA-linux-ubuntu-16.04-android/lastFailedBuild/console), based on the default logic linked above. My own Android CI doesn't build for the linux host though, only for Android armv7 [with the flags `--cross-compile-hosts=android-armv7 --skip-local-build`](https://github.com/buttaface/swift-android-sdk/blob/main/.github/workflows/sdks.yml#L134), so the host default logic chooses pthreads for the Android host and [fails for armv7 only](https://github.com/buttaface/swift-android-sdk/runs/6933354497?check_suite_focus=true#step:6:1057), as [pthreads doesn't support 32-bit here](https://github.com/apple/swift/pull/59287/files#diff-01c32d9dd91f3dff021d98ee0357a5e7c9d0c059eaab416065de0dbe54d2040aR134).

In this case, it so happens that this issue is getting the Android build to pass, but in general, it's not a good idea to build all stdlibs against only one type of threading library.

### PR DESCRIPTION
**Pull request by al45tair**
These were *supposed* to work for cross-compilation, but because of an error in `Threading.cmake`, they didn't.  Also, the test rig should be using `threading_package_name()` rather than `threading_package_default()`, so that any user override takes precedence.

Fixes #59568.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -398,9 +398,9 @@ foreach(SDK ${SWIFT_SDKS})
         if("${SWIFT_SDK_${SDK}_THREADING_PACKAGE}")
           list(APPEND LIT_ARGS "--param" "threading=${SWIFT_SDK_${SDK}_THREADING_PACKAGE}")
         else()
-          threading_package_default("${SDK}" SWIFT_THREADING_PACKAGE)
+          threading_package_name("${SDK}" _threading_package)
           list(APPEND LIT_ARGS "--param"
-            "threading=${SWIFT_THREADING_PACKAGE}")
+            "threading=${_threading_package}")
```
[the remark this answers is not in the document]
Comment (al45tair) [reply test/CMakeLists.txt:403]: Should be fixed now.

**Comment by al45tair** [comment]
@swift-ci Please test

**Review by kateinoigakukun — APPROVED** [review]

**Comment by al45tair** [comment]
I have a few more changes to go into this one yet, I think.

**Comment by al45tair** [comment]
@swift-ci Plese test

File: CMakeLists.txt [hunk 2]
```diff
@@ -1054,7 +1044,6 @@ if(SWIFT_BUILD_STDLIB OR SWIFT_BUILD_SDK_OVERLAY)
   message(STATUS "  Leak Detection Checker Entrypoints: ${SWIFT_RUNTIME_ENABLE_LEAK_CHECKER}")
   message(STATUS "")
 
-  message(STATUS "Threading Package:                  ${SWIFT_THREADING_PACKAGE}")
```
Comment (buttaface) [inline CMakeLists.txt:1057]: Please replace this info output elsewhere on a per-SDK basis, say [in `_report_sdk()`](https://github.com/apple/swift/blob/main/cmake/modules/SwiftConfigureSDK.cmake#L12) if possible.

**Comment by buttaface** [comment]
I think this will work, but it is setting the threading library very late in the configuration process and at each location, which can get hairy.

Do you mind rewriting this to follow [the established process of setting SDK-specific variables in the `configure_sdk_*()` macros instead](https://github.com/apple/swift/blob/main/cmake/modules/SwiftConfigureSDK.cmake#L138)? That way it's set once in a SDK-specific global variable that you can print out in `_report_sdk()`, which will be less error-prone over time.

**Pull request closed without merging**